### PR TITLE
Move registry

### PR DESCRIPTION
**Pull request by capitancambio**

### CODE REVIEW

File: src/tentaclio/registries/stream_handler_registry.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        return _WriterContextManager(writer)
+
+
+class _StreamRegistryHolder:
+    """Module level singleton."""
+
+    instance: ClassVar[StreamHandlerRegistry] = StreamHandlerRegistry()
+
+
+def get_stream_handler_registry() -> StreamHandlerRegistry:
```
Comment (ig248) [inline src/tentaclio/registries/stream_handler_registry.py]: Why not make `StreamHandlerRegistry` itself a singleton and avoid the factory? e.g. `__new__` could return `cls.__theoneclassinstance`

File: src/tentaclio/__init__.py [hunk 2]
```diff
@@ -15,26 +15,27 @@
 from .api import *  # noqa
 from .protocols import *  # noqa
 from .urls import *  # noqa
+from .registries import get_stream_handler_registry
 
 
 # Local files
-URL.register_handler("", LocalFileHandler())
-URL.register_handler("file", LocalFileHandler())
+get_stream_handler_registry().register("", LocalFileHandler())
```
Comment (ig248) [inline src/tentaclio/__init__.py]: Would it be more concise to call the factory/singleton constructor once and assign it to a variable, even though it's the same object? - also see comment below

File: src/tentaclio/registries/stream_handler_registry.py (hunk 1 above)
Comment (capitancambio) [reply src/tentaclio/registries/stream_handler_registry.py]: so the responsablity of storing the singleton/actual functionality is decoupled. Single responsibility principle. namely testing is cleaner as you don't have to get into the internal of the registry to reset the instance

File: src/tentaclio/__init__.py (hunk 2 above)
Comment (capitancambio) [reply src/tentaclio/__init__.py]: I'll do that

**Pull request merged**